File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        min_val = prices[0]
        max_profit = 0
        for i in prices:
            min_val = min_val if i > min_val else i
            max_profit = max_profit if max_profit > i - min_val else i - min_val
        return max_profit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min_val = prices[0];
        int max_profit = 0;
        for(int i = 0; i < prices.length; i++){
            min_val = (min_val < prices[i])?min_val:prices[i];
            max_profit = (max_profit > prices[i] - min_val)?max_profit: prices[i] - min_val;
        }
        return max_profit;
    }
}